project/Extensions.scala

```diff
@@ -0,0 +1,15 @@
+import scala.scalanative.build._
+import scala.util.Properties
+
+object Extensions {
+
+  implicit class ImplicitNativeConfig(private val cfg: NativeConfig)
+      extends AnyVal {
+    def withGC: NativeConfig =
+      if (Properties.isWin) cfg.withGC(GC.commix) else cfg.withGC(GC.immix)
+    def forTest: NativeConfig = withGC.withMode(Mode.releaseFast)
+    def forRelease: NativeConfig = withGC.withMode(Mode.releaseFull)
+      .withLTO(LTO.thin)
```

Comment:
The `LTO.thin` seems unstable on mac os.

Reply:
@He-Pin  do you know if something is known to have a problem on windows? it keeps hanging, so i am trying different options.

Reply:
I only encounter https://github.com/scala-native/scala-native/issues/4341 this problem